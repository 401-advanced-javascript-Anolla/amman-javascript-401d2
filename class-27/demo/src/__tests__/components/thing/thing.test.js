import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Thing from '../../../components/thing';

describe('<Thing/>', () => {
  it('is alive at application start', () => {
    const thing = shallow(<Thing />);
    // expect(thing.find('#me').exists()).toBeTruthy();
    // expect(thing.find('p').exists()).toBeTruthy();
    expect(thing.find('section aside h2').exists()).toBeTruthy();
  });
  it('changes state on click', () => {
    const thing = mount(<Thing />);
    const button = thing.find('button');
    button.simulate('click');
    expect(thing.state.stuff).toBeFalsy();
  });

  it('Renders correctly', () => {
    const tree = renderer.create(<Thing />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
