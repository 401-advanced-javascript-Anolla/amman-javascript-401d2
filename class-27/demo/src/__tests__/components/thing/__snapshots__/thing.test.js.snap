// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thing/> Renders correctly 1`] = `
<section>
  <p
    id="me"
  >
    Current Val: 
    true
  </p>
  <aside>
    <h2>
      hi
    </h2>
  </aside>
  <button
    onClick={[Function]}
  >
    Change
  </button>
</section>
`;
